analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -131,6 +131,13 @@ private ExplodedNode ProcessBranch(ExplodedNode node, ControlFlowBranch branch)
         {
             return FromFinally(node.FinallyPoint.CreateNext());     // Redirect from finally back to the original place (or outer finally on the same branch)
         }
+        else if (branch.Semantics == ControlFlowBranchSemantics.Rethrow
```

Comment:
This PR changes the engine itself => it needs a dedicated UT in `RoslynSymbolicExecutionTest.Branching.cs` just for this